packages/blade/src/components/Accordion/AccordionContext.tsx

```diff
@@ -0,0 +1,13 @@
+import { createContext, useContext } from 'react';
+
+type AccordionContextState = {
+  expandedIndex?: number;
+};
+
+const AccordionContext = createContext<AccordionContextState>({
+  expandedIndex: undefined,
+});
+
+const useAccordionContext = (): AccordionContextState => useContext(AccordionContext);
```

Comment:
i think all our custom context hooks shall be like this
```
export const useModalContext = () => {
  const modalContext = React.useContext(ModalContext);
  if (modalContext === undefined) {
    throw new Error('useModalContext must be used within a ModalConextProvider');
  }
  return modalContext;
};
```

this way we'll have well defined error feedback mechanisms

Reply:
Added in #1296